PSPs/drafts/psp-34.md

```diff
@@ -0,0 +1,914 @@
+      "label": "operator",
+      "type": {
+        "displayName": [
+          "AccountId"
+        ],
+        "type": "AccountId"
+      }
+    },
+    {
+      "label": "approved",
+      "type": {
+        "displayName": [
+        "bool"
+        ],
+        "type": "bool"
+      }
+    }
+  ],
+  "docs": [
+    "Approves or disapproves the operator for all tokens of the caller.",
+    "",
+    "On success a `ApprovalForAll` event is emitted.",
+    "",
+    "# Errors",
+    "",
+    "Returns `SelfApprove` error if it is self approve."
+  ],
+  "mutates": true,
+  "label": "PSP34::set_approval_for_all",
+  "payable": false,
+  "returnType": {
+    "displayName": [
+      "Result"
+    ],
+    "type": 1
+  },
+  "selector": "0xf8938214"
+}
+```
+
+##### **approve**(to: AccountId, id: Id) ➔ Result<(), PSP34Error>
+Selector: `0x1932a8b0` - first 4 bytes of `blake2b_256("PSP34::approve")`
+```json
+{
+  "args": [
+    {
+      "label": "to",
+      "type": {
+        "displayName": [
+          "AccountId"
+        ],
+        "type": "AccountId"
+      }
+    },
+    {
+      "label": "id",
+      "type": {
+        "displayName": [
+           "Id"
+        ],
+        "type": "Id"
+      }
+    }
+  ],
+  "docs": [
+    "Approves the account to transfer the specified token on behalf of the caller.",
+    "",
+    "On success a `Approval` event is emitted.",
+    "",
+    "# Errors",
+    "",
+    "Returns `SelfApprove` error if it is self approve.",
+    "",
+    "Returns `NotApproved` error if caller is not owner of `id`."
+  ],
+  "mutates": true,
+  "label": "PSP34::approve",
+  "payable": false,
+  "returnType": {
+    "displayName": [
+      "Result"
+    ],
+    "type": 1
+  },
+  "selector": "0x1932a8b0"
+}
+```
+
+##### **transfer**(to: AccountId, id: Id, data: [u8]) ➔ Result<(), PSP34Error>
+Selector: `0x3128d61b` - first 4 bytes of `blake2b_256("PSP34::transfer")`
+```json
+{
+  "args": [
+    {
+      "label": "to",
+      "type": {
+        "displayName": [
+          "AccountId"
+        ],
+        "type": "AccountId"
+      }
+    },
+    {
+      "label": "id",
+      "type": {
+        "displayName": [
+           "Id"
+        ],
+        "type": "Id"
+      }
+    },
+    {
+      "label": "data",
+      "type": {
+        "displayName": [
+          "[u8]"
+        ],
+        "type": "[u8]"
+      }
+    }
+  ],
+  "docs": [
+    "Transfer approved or owned token from caller.",
+    "",
+    "On success a `Transfer` event is emitted.",
+    "",
+    "# Errors",
+    "",
+    "Returns `TokenNotExists` error if `id` is not exist.",
+    "",
+    "Returns `SafeTransferCheckFailed` error if `to` doesn't accept transfer."
+  ],
+  "mutates": true,
+  "label": "PSP34::transfer",
+  "payable": false,
+  "returnType": {
+    "displayName": [
+      "Result"
+    ],
+    "type": 1
+  },
+  "selector": "0x3128d61b"
+}
+```
+
+##### **transfer_from**(from: AccountId, to: AccountId, id: Id, data: [u8]) ➔ Result<(), PSP34Error>
```

Comment:
```suggestion
##### **transfer_from**(from: AccountId, to: AccountId, id: Id, data: Vec<u8>) ➔ Result<(), PSP34Error>
```

Reply:
We used `[u8]` instead of `Vec<u8>` because it is not standard for Rust, it is standard for any language. `[u8]` - is only array of bytes =)